statsmodels/regression/recursive_ls.py

```diff
@@ -0,0 +1,695 @@
+        start = offset = 0
+        end = offset + spec.k_exog
+        out = Bunch(
+            filtered=self.filtered_state[start:end],
+            filtered_cov=self.filtered_state_cov[start:end, start:end],
+            smoothed=None, smoothed_cov=None,
+            offset=offset
+        )
+        if self.smoothed_state is not None:
+            out.smoothed = self.smoothed_state[start:end]
+        if self.smoothed_state_cov is not None:
+            out.smoothed_cov = (
+                self.smoothed_state_cov[start:end, start:end])
+        return out
+
+    @cache_readonly
+    def recursive_resid(self):
+        """
+        Recursive residuals
+
+        Returns
+        -------
+        recursive_resid : array_like
+            An array of length `nobs` holding the recursive
+            residuals.
+
+        Notes
+        -----
+        The first `k_exog` residuals are typically unreliable due to
+        initialization.
+
+        """
+        # See Harvey (1989) section 5.4; he defines the standardized
+        # innovations in 5.4.1, but they have non-unit variance, whereas
+        # the standardized forecast errors assume unit variance. To convert
+        # to Harvey's definition, we need to multiply by the standard
+        # deviation.
+        return (self.filter_results.standardized_forecasts_error.squeeze() *
+                self.filter_results.obs_cov[0, 0]**0.5)
+
+    @cache_readonly
+    def cusum(self):
+        r"""
+        Cumulative sum of standardized recursive residuals statistics
+
+        Returns
+        -------
+        cusum : array_like
+            An array of length `nobs - k_exog` holding the
+            CUSUM statistics.
+
+
+        Notes
+        -----
+        The CUSUM statistic takes the form:
+
+        .. math::
+
+            W_t = \frac{1}{\hat \sigma} \sum_{j=k+1}^t w_j
+
+        where :math:`w_j` is the recursive residual at time :math:`j` and
+        :math:`\hat \sigma` is the estimate of the standard deviation
+        from the full sample.
+
+        Excludes the first `k_exog` datapoints.
+
+        Due to differences in the way :math:`\hat \sigma` is calculated, the
+        output of this function differs slightly from the output in the
+        R package strucchange and the Stata contributed .ado file cusum6. The
+        calculation in this package is consistent with the description of
+        Brown et al. (1975)
+
+        References
+        ----------
+        .. [1] Brown, R. L., J. Durbin, and J. M. Evans. 1975.
+           "Techniques for Testing the Constancy of
+           Regression Relationships over Time."
+           Journal of the Royal Statistical Society.
+           Series B (Methodological) 37 (2): 149-92.
+
+        """
+        llb = self.loglikelihood_burn
+        return (np.cumsum(self.recursive_resid[self.loglikelihood_burn:]) /
+                np.std(self.recursive_resid[llb:], ddof=1))
+
+    @cache_readonly
+    def cusum_squares(self):
```

Comment:
I don't know a good name that would also include resid or squared resid. Maybe fine the way it is.